appcues/src/test/java/com/appcues/action/appcues/DelayActionTest.kt

```diff
@@ -0,0 +1,52 @@
+package com.appcues.action.appcues
+
+import com.appcues.AppcuesScopeTest
+import com.appcues.rules.TestScopeRule
+import com.google.common.truth.Truth.assertThat
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.test.runTest
+import org.junit.Rule
+import org.junit.Test
+import kotlin.time.Duration.Companion.milliseconds
+import kotlin.time.ExperimentalTime
+import kotlin.time.measureTime
+
+@ExperimentalTime
+internal class DelayActionTest : AppcuesScopeTest {
+
+    @get:Rule
+    override val scopeRule = TestScopeRule()
+
+    @Test
+    fun `close SHOULD have expected type name`() {
+        assertThat(DelayAction.TYPE).isEqualTo("@appcues/delay")
+    }
+
+    @Test
+    fun `delay SHOULD call delay with given value of 300ms `() = runTest {
+        // GIVEN
+        val action = DelayAction(mapOf("duration" to 300))
+
+        launch {
+            val workDuration = testScheduler.timeSource.measureTime {
```

Comment:
TIL `measureTime` cool - and it is always the exact ms of the delay? thats sort of amazing but useful here

Reply:
yes. [reference](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-test/#:~:text=To%20test%20regular%20suspend%20functions,of%20delaying%20%2F%2F%20...%20%7D)